### PR TITLE
support for more formats to work with RichDate

### PR DESCRIPTION
**Pull request by afsalthaj**

### CODE REVIEW

File: scalding-date/src/main/scala/com/twitter/scalding/DateOps.scala [hunk 1]
```diff
@@ -38,14 +42,19 @@ object DateOps extends java.io.Serializable {
   }
 
   private[scalding] object Format {
+    private val normalDate = """\d{4}\d{2}\d{2}"""
```
Comment (ianoc) [inline scalding-date/src/main/scala/com/twitter/scalding/DateOps.scala]: can we put a better name on this? dateWithoutSep maybe?

File: scalding-date/src/main/scala/com/twitter/scalding/DateOps.scala [hunk 2]
```diff
@@ -59,8 +68,9 @@ object DateOps extends java.io.Serializable {
    * Return the guessed format for this datestring
    */
   private[scalding] def getFormatObject(s: String): Option[Format] = {
-    val formats: List[Format] = List(Format.DATE_WITH_DASH, Format.DATEHOUR_WITH_DASH,
-      Format.DATETIME_WITH_DASH, Format.DATETIME_HMS_WITH_DASH, Format.DATETIME_HMSM_WITH_DASH)
+    val formats: List[Format] = List(Format.DATE_WITHOUT_DASH, Format.DATE_WITH_DASH, Format.DATEHOUR_WITH_DASH,
```
Comment (ianoc) [inline scalding-date/src/main/scala/com/twitter/scalding/DateOps.scala]: given the length of this list now i think it makes sense to do a 1 per line style

List(
 entry1,
 entry2,
 entry3
)

**Comment by ianoc** [comment]
Overall looks good to me, just some minor style nits

File: scalding-date/src/main/scala/com/twitter/scalding/DateOps.scala (hunk 1 above)
Comment (afsalthaj) [inline scalding-date/src/main/scala/com/twitter/scalding/DateOps.scala]: Sure. my bad!!.. I will change that..

**Comment by afsalthaj** [comment]
@ianoc  Thanks for pointing out...
Could you please point out in general on those minor style nits... Never want to change the existing styles the project has been following

File: scalding-date/src/main/scala/com/twitter/scalding/DateOps.scala (hunk 2 above)
Comment (afsalthaj) [inline scalding-date/src/main/scala/com/twitter/scalding/DateOps.scala]: I will make this change

**Comment by afsalthaj** [comment]
@ianoc  pls see if this ok for you to merge

File: scalding-date/src/main/scala/com/twitter/scalding/DateOps.scala [hunk 3]
```diff
@@ -38,14 +42,19 @@ object DateOps extends java.io.Serializable {
   }
 
   private[scalding] object Format {
+    private val dateWithoutSep = """\d{4}\d{2}\d{2}"""
```
Comment (johnynek) [inline scalding-date/src/main/scala/com/twitter/scalding/DateOps.scala]: isn't this just `\d{8}`?

File: scalding-date/src/main/scala/com/twitter/scalding/DateOps.scala [hunk 4]
```diff
@@ -38,14 +42,19 @@ object DateOps extends java.io.Serializable {
...
     private val date = """\d{4}-\d{2}-\d{2}"""
     private val sep = """(T?|\s*)"""
     private val emptyBegin = """^\s*"""
     private val emptyEnd = """\s*$"""
 
+    case object DATE_WITHOUT_DASH extends Format(DateOps.DATE_WITHOUT_DASH, new Regex(emptyBegin + dateWithoutSep + emptyEnd))
     case object DATE_WITH_DASH extends Format(DateOps.DATE_WITH_DASH, new Regex(emptyBegin + date + emptyEnd))
+    case object DATEHOUR_WITHOUT_DASH extends Format(DateOps.DATEHOUR_WITHOUT_DASH, new Regex(emptyBegin + dateWithoutSep + """\d\d""" + emptyEnd))
```
Comment (johnynek) [inline scalding-date/src/main/scala/com/twitter/scalding/DateOps.scala]: can you use `\d{10}` here rathe the the define. Similarly below `\d{12}` etc...

File: scalding-date/src/main/scala/com/twitter/scalding/DateOps.scala [hunk 5]
```diff
@@ -38,14 +42,19 @@ object DateOps extends java.io.Serializable {
   }
 
   private[scalding] object Format {
+    private val dateWithoutSep = """\d{4}\d{2}\d{2}"""
     private val date = """\d{4}-\d{2}-\d{2}"""
     private val sep = """(T?|\s*)"""
     private val emptyBegin = """^\s*"""
     private val emptyEnd = """\s*$"""
 
+    case object DATE_WITHOUT_DASH extends Format(DateOps.DATE_WITHOUT_DASH, new Regex(emptyBegin + dateWithoutSep + emptyEnd))
```
Comment (johnynek) [inline scalding-date/src/main/scala/com/twitter/scalding/DateOps.scala]: lets just use `\d{8}` directly here and delete above.

File: scalding-date/src/main/scala/com/twitter/scalding/DateOps.scala [hunk 6]
```diff
@@ -59,8 +68,17 @@ object DateOps extends java.io.Serializable {
    * Return the guessed format for this datestring
    */
   private[scalding] def getFormatObject(s: String): Option[Format] = {
-    val formats: List[Format] = List(Format.DATE_WITH_DASH, Format.DATEHOUR_WITH_DASH,
-      Format.DATETIME_WITH_DASH, Format.DATETIME_HMS_WITH_DASH, Format.DATETIME_HMSM_WITH_DASH)
+    val formats: List[Format] = List(
+      Format.DATE_WITHOUT_DASH,
```
Comment (johnynek) [inline scalding-date/src/main/scala/com/twitter/scalding/DateOps.scala]: please put `WITH_DASH` first and all the new formats after so existing use cases are not slowed down.

File: scalding-date/src/test/scala/com/twitter/scalding/DateTest.scala [hunk 7]
```diff
@@ -27,13 +27,23 @@ class DateTest extends WordSpec {
     "implicitly convert strings" in {
       val rd1: RichDate = "2011-10-20"
       val rd2: RichDate = "2011-10-20"
+      val rd3: RichDate = "20111020"
+      val rd4: RichDate = "20111020"
       assert(rd1 === rd2)
+      assert(rd3 === rd4)
```
Comment (johnynek) [inline scalding-date/src/test/scala/com/twitter/scalding/DateTest.scala:33]: add `rd1 === rd3`

File: scalding-date/src/main/scala/com/twitter/scalding/DateOps.scala (hunk 6 above)
Comment (afsalthaj) [inline scalding-date/src/main/scala/com/twitter/scalding/DateOps.scala]: sure

File: scalding-date/src/test/scala/com/twitter/scalding/DateTest.scala (hunk 7 above)
Comment (afsalthaj) [inline scalding-date/src/test/scala/com/twitter/scalding/DateTest.scala:33]: sure...il add this

File: scalding-date/src/main/scala/com/twitter/scalding/DateOps.scala (hunk 3 above)
Comment (afsalthaj) [inline scalding-date/src/main/scala/com/twitter/scalding/DateOps.scala]: Yeah...il do that.. My bad !!

**Comment by afsalthaj** [comment]
I have made the changes required.

**Comment by johnynek** [comment]
Thanks!

**Pull request merged**